File: app/src/main/java/com/zionhuang/music/youtube/SuggestionAPIService.java
package com.zionhuang.music.youtube;

import io.reactivex.rxjava3.core.Observable;
import retrofit2.http.GET;
import retrofit2.http.Query;

public interface SuggestionAPIService {
    @GET("/complete/search?client=firefox&ds=yt")
    Observable<SuggestionResult> suggest(@Query("q") String query);
}

File: app/src/main/java/com/zionhuang/music/youtube/SuggestionAPIService.kt
package com.zionhuang.music.youtube

import retrofit2.Response
import retrofit2.http.GET
import retrofit2.http.Query

interface SuggestionAPIService {
    @GET("/complete/search?client=firefox&ds=yt")
    suspend fun suggest(@Query("q") query: String): Response<SuggestionResult>
}